docs/src/devdocs/special_datastructures.md

```diff
@@ -0,0 +1,14 @@
+# Special data structures
+
+## `ArrayOfVectorViews`
```

Comment:
Could the obvious question of "why not use ArraysOfArrays" be answered shortly in this section as well?